### PR TITLE
Rename _MatchingEngine module to _RegexParser

### PR DESCRIPTION
**Pull request by rxwei**
As the _MatchingEngine module no longer contains the matching engine, this patch renames this module to describe its role more accurately. Because this module primarily contains the AST and the regex parsing logic, I propose we rename it to "_RegexParser".

Also renames the ExperimentalRegex module in SwiftCompilerSources to _RegexParser for consistency. This would prevent errors if sources in _RegexParser used qualified lookup with the module name.

### CODE REVIEW

**Comment by rxwei** [comment]
apple/swift-experimental-string-processing#232
@swift-ci please test

**Review by hamishknight — APPROVED** [review]

**Comment by rxwei** [comment]
TODO: Rename to `_RegexParser`.

**Comment by rxwei** [comment]
apple/swift-experimental-string-processing#235
@swift-ci please test

**Comment by rxwei** [comment]
apple/swift-experimental-string-processing#235
@swift-ci please test

**Comment by rxwei** [comment]
apple/swift-experimental-string-processing#235
@swift-ci please test

**Comment by rxwei** [comment]
apple/swift-experimental-string-processing#235
@swift-ci please test

File: SwiftCompilerSources/Sources/ExperimentalRegex/CMakeLists.txt [hunk 1]
```diff
@@ -1,21 +0,0 @@
...
-foreach(source ${_LIBSWIFT_EXPERIMENTAL_REGEX_SOURCES})
-  file(TO_CMAKE_PATH "${source}" source)
-  list(APPEND LIBSWIFT_EXPERIMENTAL_REGEX_SOURCES ${source})
-endforeach()
-message(STATUS "Using Experimental String Processing library for libswift ExperimentalRegex (${EXPERIMENTAL_STRING_PROCESSING_SOURCE_DIR}).")
-
-add_swift_compiler_module(ExperimentalRegex
-  "${LIBSWIFT_EXPERIMENTAL_REGEX_SOURCES}"
-  Regex.swift)
-
```
Comment (rintaro) [inline SwiftCompilerSources/Sources/ExperimentalRegex/CMakeLists.txt:21]: Did you accidentally deleted this and the `Regex.swift` source?

**Comment by rintaro** [comment]
I assume `_RegexParser` in stdlib is visible and importable like `_Concurrency`, right? If so I have a mild concern where, in the future, `import _RegexParser` in `PassRegistration.swift` would be ambiguous. Maybe it might be safe to have separate names between stdlib and compiler. What do you think?

**Comment by rxwei** [comment]
> I assume _RegexParser in stdlib is visible and importable like _Concurrency, right?

It's not supposed to be importable.  We plan to strip out the the module files when installing _RegexParser.

File: SwiftCompilerSources/Sources/ExperimentalRegex/CMakeLists.txt (hunk 1 above)
Comment (rxwei) [reply SwiftCompilerSources/Sources/ExperimentalRegex/CMakeLists.txt:21]: Good catch! Forgot to git add.

**Comment by rxwei** [comment]
apple/swift-experimental-string-processing#235
@swift-ci please test

**Review by rintaro — APPROVED** [review]
Richard and I talked about this and I am now convinced that this won't be a much problem.

**Pull request merged**